bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -169,22 +168,21 @@ public FutureValue scheduleLocal(Object[] params, BFunctionPointer<?, ?> fp, Str
         return future;
     }
 
-    public FutureValue scheduleToObjectGroup(BObject object, Object[] params, Function function, Strand parent,
+    public FutureValue scheduleToObjectGroup(Object[] params, Function function, Strand parent,
                                              Callback callback, Map<String, Object> properties, Type returnType,
                                              String strandName, StrandMetadata metadata) {
         FutureValue future = createFuture(parent, callback, properties, returnType, strandName, metadata);
         params[0] = future.strand;
         SchedulerItem item = new SchedulerItem(function, params, future);
         future.strand.schedulerItem = item;
         totalStrands.incrementAndGet();
-        ItemGroup group = objectGroups.compute(object, (o, groupInMap) -> {
-            if (groupInMap == null) {
-                return new ItemGroup(item);
-            } else {
-                groupInMap.add(item);
-                return groupInMap;
-            }
-        });
+        ItemGroup group = objectGroup.get();
+        if (group == null) {
+            group = new ItemGroup(item);
```

Comment:
The `objectGroup` need to initialised here as we can't create an `ItemGroup` without a `ScedulerItem`